lib/websocket_proxy.rb

```diff
@@ -15,7 +15,7 @@ def initialize(env, console)
       # Hijack the socket from the Rack middleware
       @ws = env['rack.hijack'].call
       # Set up the socket client for the proxy
-      @sock = TCPSocket.open(@console.host_name, @console.port)
+      Timeout.timeout(30) { @sock = TCPSocket.open(@console.host_name, @console.port) }
```

Comment:
Perhaps, I believe @skateman was telling me that he need to confirm the problem recreation steps before we can verify and merge a solution like this.  Otherwise, we're just guessing if it fixes it.